cmd/auth.go

```diff
@@ -0,0 +1,119 @@
+package cmd
+
+import (
+	"errors"
+	"syscall"
+
+	"github.com/jsdelivr/globalping-cli/globalping"
+	"github.com/spf13/cobra"
+)
+
+func (r *Root) initAuth() {
+	authCmd := &cobra.Command{
+		Use:   "auth",
+		Short: "Authenticate with the Globalping API",
+		Long:  "Authenticate with the Globalping API for higher measurements limits.",
+	}
+
+	loginCmd := &cobra.Command{
+		RunE:  r.RunAuthLogin,
+		Use:   "login",
+		Short: "Log in to your Globalping account",
+		Long:  `Log in to your Globalping account for higher measurements limits.`,
+	}
+
+	loginFlags := loginCmd.Flags()
+	loginFlags.Bool("with-token", false, "authenticate with a token read from stdin instead of the default browser-based flow")
+
+	statusCmd := &cobra.Command{
+		RunE:  r.RunAuthStatus,
+		Use:   "status",
+		Short: "Check the current authentication status",
+		Long:  `Check the current authentication status.`,
+	}
+
+	logoutCmd := &cobra.Command{
+		RunE:  r.RunAuthLogout,
+		Use:   "logout",
+		Short: "Log out from your Globalping account",
+		Long:  `Log out from your Globalping account.`,
+	}
+
+	authCmd.AddCommand(loginCmd)
+	authCmd.AddCommand(statusCmd)
+	authCmd.AddCommand(logoutCmd)
+
+	r.Cmd.AddCommand(authCmd)
+}
+
+func (r *Root) RunAuthLogin(cmd *cobra.Command, args []string) error {
+	var err error
+	withToken := cmd.Flags().Changed("with-token")
+	if withToken {
+		return r.loginWithToken()
+	}
+	url := r.client.Authorize(func(e error) {
+		defer func() {
+			r.cancel <- syscall.SIGINT
+		}()
+		if e != nil {
+			err = e
+			return
+		}
+		r.printer.Println("You are now authenticated")
+	})
+	r.printer.Println("Please visit the following URL to authenticate:")
```

Comment:
Can we also automatically open this in the default browser? (ignoring any errors if that fails)